tests/integration/awslambda/test_lambda_integration.py

```
@@ -422,6 +597,225 @@ def verify_failure_received():
+        item_to_put1,
+        item_to_put2,
+    ):
+
+        function_name = f"lambda_func-{short_uid()}"
+        table_name = f"test-table-{short_uid()}"
+        event_source_uuid = None
+        max_retries = 50
+
+        try:
+
+            create_lambda_function(
+                handler_file=TEST_LAMBDA_PYTHON_ECHO,
+                func_name=function_name,
+                runtime=LAMBDA_RUNTIME_PYTHON37,
+                role=lambda_su_role,
+            )
+            dynamodb_create_table(table_name=table_name, partition_key="id")
+            _await_dynamodb_table_active(dynamodb_client, table_name)
+            stream_arn = dynamodb_client.update_table(
+                TableName=table_name,
+                StreamSpecification={"StreamEnabled": True, "StreamViewType": "NEW_AND_OLD_IMAGES"},
+            )["TableDescription"]["LatestStreamArn"]
+            wait_for_dynamodb_stream_ready(stream_arn)
+            event_source_mapping_kwargs = {
+                "FunctionName": function_name,
+                "BatchSize": 1,
+                "StartingPosition": "TRIM_HORIZON",
+                "EventSourceArn": stream_arn,
+                "MaximumBatchingWindowInSeconds": 1,
+                "MaximumRetryAttempts": 1,
+            }
+            event_source_mapping_kwargs.update(
+                FilterCriteria={
+                    "Filters": [
+                        {"Pattern": json.dumps(filter)},
+                    ]
+                }
+            )
+            try:
+                event_source_uuid = lambda_client.create_event_source_mapping(
+                    **event_source_mapping_kwargs
+                )["UUID"]
+            except ClientError as ex:
+                if filter:
+                    raise ex
+                # json.loads serializes None to "FilterCriteria": {"Filters": [{"Pattern": "null"}]}}
+                assert ex.response["Error"]["Code"] == INVALID_PARAMETER_VALUE_EXCEPTION
+                return
```

NIT: I think we can remove the entire "except" block - there are no input parameters anymore that should throw an expected exception.